eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -171,10 +171,9 @@ class Router(val nodeParams: NodeParams, watcher: typed.ActorRef[ZmqWatcher.Comm
     case Event(TickPruneStaleChannels, d) =>
       stay() using StaleChannels.handlePruneStaleChannels(d, nodeParams.db.network, nodeParams.currentBlockHeight)
 
-    case Event(ExcludeChannel(desc@ChannelDesc(shortChannelId, nodeId, _)), d) =>
-      val banDuration = nodeParams.routerConf.channelExcludeDuration
-      log.info("excluding shortChannelId={} from nodeId={} for duration={}", shortChannelId, nodeId, banDuration)
-      context.system.scheduler.scheduleOnce(banDuration, self, LiftChannelExclusion(desc))
+    case Event(ExcludeChannel(desc, duration_opt), d) =>
+      log.info("excluding shortChannelId={} from nodeId={} for duration={}", desc.shortChannelId, desc.a, duration_opt.getOrElse("n/a"))
+      duration_opt.foreach(banDuration => context.system.scheduler.scheduleOnce(banDuration, self, LiftChannelExclusion(desc)))
```

Comment:
I would make `routerConf.channelExcludeDuration` a `Duration`, which can either be a `FiniteDuration` or `InfiniteDuration`. We can then define the default duration as `infinite` in `eclair.conf` and leave `ExcludeChannel` and `PaymentLifecycle` untouched.

```suggestion
    case Event(ExcludeChannel(desc), d) =>
      val banDuration = nodeParams.routerConf.channelExcludeDuration
      log.info("excluding shortChannelId={} from nodeId={} for duration={}", desc.shortChannelId, desc.a, banDuration)
      banDuration match {
        case duration: FiniteDuration => context.system.scheduler.scheduleOnce(duration, self, LiftChannelExclusion(desc))
        case _: Duration.Infinite => () // permanent ban, can still be manually lifted by sending a LiftChannelExclusion
      }
```

Reply:
> I would make routerConf.channelExcludeDuration a Duration, which can either be a FiniteDuration or InfiniteDuration. We can then define the default duration as infinite in eclair.conf and leave ExcludeChannel and PaymentLifecycle untouched.

But does it really make sense? I don't think the errors in `PaymentLifecycle` should ever trigger an infinite ban, so it would be a footgun to let node operators set this to `infinite` in `eclair.conf`, wouldn't it?

The goal of the issue I linked is to let plugins ban channels for the duration they'd like (infinite or not), independently of what `PaymentLifecycle` may do. It lets plugins experiment with graph pruning heuristics without requiring big changes in `eclair`.

Reply:
> The goal of the issue I linked is to let plugins ban channels for the duration they'd like (infinite or not), independently of what PaymentLifecycle may do. It lets plugins experiment with graph pruning heuristics without requiring big changes in eclair.

Alright, then we can:
- keep `routeParams.channelExcludeDuration` as `FiniteDuration` (altough node operators can still set it to 1 year)
- add a `duration_opt: Option[Duration]` to `ExcludeChannel`. The semantics of the `Option` changes from "emulating an infite duration" to "override the default value"

Reply:
> add a duration_opt: Option[Duration] to ExcludeChannel. The semantics of the Option changes from "emulating an infite duration" to "override the default value"

I tried this, but the `InfiniteDuration` is quite awkward to use, we need an explicit check on `isFinite`...it feels more natural to provide `None` for a permanent exclusion than to use `Some(InfiniteDuration)` don't you think? If provided, the channel is banned for that duration, if not provided the channel is banned until lifted manually.